diff -ur redmine_wysiwyg_editor.orig/app/views/redmine_wysiwyg_editor/_redmine_wysiwyg_editor_partial.html.erb redmine_wysiwyg_editor/app/views/redmine_wysiwyg_editor/_redmine_wysiwyg_editor_partial.html.erb
--- redmine_wysiwyg_editor.orig/app/views/redmine_wysiwyg_editor/_redmine_wysiwyg_editor_partial.html.erb	2021-02-06 14:13:16.000000000 +0000
+++ redmine_wysiwyg_editor/app/views/redmine_wysiwyg_editor/_redmine_wysiwyg_editor_partial.html.erb	2021-02-21 17:36:58.084461534 +0000
@@ -39,10 +39,6 @@
   margin: 0;
 }
 
-#content .jstBlock .jstTabs ul li:not(.tab-elements) {
-  display: none;
-}
-
 #content .jstBlock .jstTabs .jstElements {
   height: auto;
 }
diff -ur redmine_wysiwyg_editor.orig/assets/javascripts/redmine_wysiwyg_editor.js redmine_wysiwyg_editor/assets/javascripts/redmine_wysiwyg_editor.js
--- redmine_wysiwyg_editor.orig/assets/javascripts/redmine_wysiwyg_editor.js	2021-02-06 14:13:16.000000000 +0000
+++ redmine_wysiwyg_editor/assets/javascripts/redmine_wysiwyg_editor.js	2021-02-21 17:39:05.790576652 +0000
@@ -119,29 +119,38 @@
 
   var previewHtml = '<div class="wysiwyg-editor-preview wiki"></div>';
 
-  var modeTabHtml = '<div class="wysiwyg-editor-tab"><ul>' +
-      '<li><a href="#" data-type="text" class="active">' +
-      self._i18n[self._format] + '</a></li>' +
-      '<li><a href="#" data-type="visual">' +
-      self._i18n.visual + '</a></li>' +
-      '<li><a href="#" data-type="preview">' +
-      self._i18n.preview + '</a></li>' +
-      '</ul></div>';
-
   self._jstEditorTextArea = self._jstEditor.find('textarea');
 
   self._jstEditorTextArea.after(previewHtml);
-  self._jstEditor.after(editorHtml + modeTabHtml);
+  self._jstEditor.after(editorHtml);
 
   self._visualEditor = container.find('.wysiwyg-editor').hide();
   self._preview = container.find('.wysiwyg-editor-preview').hide();
-  self._modeTab = container.find('.wysiwyg-editor-tab');
 
   var jstTabs = container.find('.jstTabs');
   var jstElements = container.find('.jstElements');
 
   if (jstTabs.length > 0) {
-    self._jstElements = jstTabs;
+    self._jstElements = jstElements;
+    self._tabsUl = jstTabs[0].firstChild;
+    self._tabsUl.removeChild(self._tabsUl.firstChild);
+    self._tabsUl.removeChild(self._tabsUl.firstChild);
+
+    jsToolBar.strings['rwe-preview'] = self._i18n.preview;
+    var prvtab = new jsTab('rwe-preview', false);
+    prvtab.onclick = function(event){self._changeMode('preview');}
+    self._tabsUl.prepend(prvtab);
+
+    jsToolBar.strings['rwe-visual'] = self._i18n.visual;
+    var vistab = new jsTab('rwe-visual', false);
+    vistab.onclick = function(event) {self._changeMode('visual');}
+    self._tabsUl.prepend(vistab);
+    
+    jsToolBar.strings['rwe-text'] = self._i18n[self._format];
+    var stdtab = new jsTab('rwe-text', true);
+    stdtab.onclick = function(event) {self._changeMode('text');}
+    self._tabsUl.prepend(stdtab);
+    
     self._oldPreviewAccess = false;
     self._preview.addClass('wiki-preview');
   } else {
@@ -149,11 +158,6 @@
     self._oldPreviewAccess = true;
   }
 
-  self._modeTab.on('click', 'li a', function(e) {
-    e.preventDefault();
-    self._changeMode($(this).data('type'));
-  });
-
   self._defaultMode =
     (('localStorage' in window) && (window.localStorage !== null)) ? {
       get: function() {
@@ -187,10 +191,14 @@
 
   if (!self._editor) return false;
 
-  self._modeTab.find('li a').each(function() {
-    if ($(this).data('type') === mode) $(this).addClass('active');
-    else $(this).removeClass('active');
-  });
+  for (var i = 0; i < 3; i++) {
+    var thisTab = self._tabsUl.childNodes[i].firstChild;
+    if (thisTab.classList.contains('tab-rwe-' + mode)) {
+      thisTab.classList.add('selected');
+    } else {
+      thisTab.classList.remove('selected');
+    }
+  }
 
   switch (mode) {
   case 'visual':
diff -ur redmine_wysiwyg_editor.orig/assets/stylesheets/redmine_wysiwyg_editor.css redmine_wysiwyg_editor/assets/stylesheets/redmine_wysiwyg_editor.css
--- redmine_wysiwyg_editor.orig/assets/stylesheets/redmine_wysiwyg_editor.css	2021-02-06 14:13:16.000000000 +0000
+++ redmine_wysiwyg_editor/assets/stylesheets/redmine_wysiwyg_editor.css	2021-02-21 17:34:25.130718655 +0000
@@ -3,7 +3,7 @@
 }
 
 .wysiwyg-editor-preview.wiki-preview {
-  background-color: #fafafa;
+  background-color: #ffffff;
 }
 
 .wysiwyg-editor-preview p {
